sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -145,34 +171,23 @@ protected IList<PropertyData> CollectPropertyData(IEnumerable<IPropertySymbol> p
             return allPropertyData;
         }
 
-        /**
-         * Assignments can be done either
-         * - directly via an assignment
-         * - indirectly, when passed as 'out' or 'ref' parameter
-         */
-        protected abstract IEnumerable<FieldData> FindFieldAssignments(IPropertySymbol property, Compilation compilation);
-        protected abstract IEnumerable<FieldData> FindFieldReads(IPropertySymbol property, Compilation compilation);
-        protected abstract bool ImplementsExplicitGetterOrSetter(IPropertySymbol property);
-        protected abstract bool ShouldIgnoreAccessor(IMethodSymbol accessorMethod);
-
-
         protected struct PropertyData
         {
             public PropertyData(IPropertySymbol propertySymbol, IEnumerable<FieldData> read, IEnumerable<FieldData> updated,
                 bool ignoreGetter, bool ignoreSetter)
             {
                 PropertySymbol = propertySymbol;
-                ReadFields = read;
-                UpdatedFields = updated;
+                ReadFields = read.ToImmutableArray();
```

Comment:
This is nice but it would make sense in this case that both FindFieldAssignments and FindFieldReads to return ImmutableArray<T>. What do you think?

Reply:
I considered that and decided to go with this version to remove unneeded duplication. It's PropertyData that really needs ImmutableArray, so it make sense to me that this is the single point responsible of doing so.